docs: explain glob patterns (#32728)
microsoft/playwright: docs/src/network.md

});
```

## Glob URL patterns

> Playwright uses glob patterns for URL matching in network interception methods like [`method: Page.route`] or [`method: Page.waitForResponse`]. These glob patterns are more specific than typical file globs and offer features similar to regular expressions:

Reviewer: ```suggestion
Playwright uses glob patterns for URL matching in network interception methods like [`method: Page.route`] or [`method: Page.waitForResponse`]. These glob patterns are more limited than typical file globs:
```

1. Asterisks:
   - A single `*` matches any characters except `/`
   - A double `**` matches any characters including `/`
1. Question mark `?` matches any single character except `/`
1. Character sets `[]` work like [RegExp character sets](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions/Character_classes#types):
   - `[abc]` matches any single character from the set (a, b, or c)
   - `[a-z]` matches any single character in the range a to z

> 1. Alternatives `{}` work like a logical OR:

Reviewer: These don't support nesting in our case. I'd not include this at all.
